rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_components_associated_with_each_swh_bat.py

```diff
@@ -0,0 +1,108 @@
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_building_segment_swh_bat import (
+    get_building_segment_swh_bat,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_energy_required_to_heat_swh_use import (
+    get_energy_required_to_heat_swh_use,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_swh_uses_associated_with_each_building_segment import (
+    get_swh_uses_associated_with_each_building_segment,
+)
+from rct229.utils.jsonpath_utils import find_all
+from rct229.utils.pint_utils import ZERO
+from rct229.utils.utility_functions import (
+    find_exactly_one_service_water_heating_use,
+    find_exactly_one_service_water_heating_distribution_system,
+)
+
+
+def get_swh_components_associated_with_each_swh_bat(
+    rmd: dict, is_leap_year: bool
+) -> dict[str, dict]:
+    """
+    This function gets all the SWH equipment connected to a particular SWH use type.  The information is stored in a dictionary where the keys are the ServiceWaterHeatingSpaceOptions2019ASHRAE901 and values are a dictionary giving the ServiceWaterHeatingDistributionSystem, ServiceWaterHeatingEquipment, and Pumps connected to the particular use type.  There is also an entry for the total energy required to heat all SWH uses for a year: "EnergyRequired" .
+
+    Parameters
+    ----------
+    rmd: dict, RMD at RuleSetModelDescription level
+    is_leap_year: bool, default: False
+        Whether the year is a leap year or not.
+    Returns
+    -------
+    swh_and_equip_dict: A dictionary containing where the keys are the ServiceWaterHeatingSpaceOptions2019ASHRAE901 in the RMD and values are dictionaries where keys are the type of SWH equipment and values are the ids of the connected equipment.
+                        Example:  {"DORMITORY":{"SWHDistribution":["swhd1","swhd2"], "SWHHeatingEq":["swh_eq1","swh_eq2"], "Pumps":["p1"], "Tanks":["t1"], "Piping":["piping1"], "SolarThermal":[], "SWH_Uses":["swh_use1","swh_use2"], "EnergyRequired": 100000 Btu}}
+    """
+
+    swh_and_equip_dict = {}
+    for building_segment in find_all("$.buildings[*].building_segments[*]", rmd):
+        swh_bat = get_building_segment_swh_bat(rmd, building_segment["id"])
+        swh_and_equip_dict.setdefault(
+            swh_bat,
+            {
+                "swh_distribution": [],
```

Comment:
If we use the dataclass created in 
![image](https://github.com/user-attachments/assets/4095cc98-65b0-4dcf-aa3b-900d03683da0), we need to exclude some existing fields and add a new field. In this case, do we still want to reuse it? Or do we create a new dataclass? @